### PR TITLE
feat(core): Feature flag library for Nest APIs

### PR DESCRIPTION
**Pull request by eirikurn**
## What

A feature flag library for our APIs. Wraps the low level API to provide support for our user objects, consistent with the `@island.is/react/feature-flags` library.

Also includes a NestJS guard to wrap entire endpoints and resolvers behind feature flags.

[Readme](https://github.com/island-is/island.is/blob/nest-feature-flag/libs/nest/feature-flags/README.md)

## Why

To make it easier to use feature flags in our APIs.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/nest/feature-flags/src/lib/feature-flag.config.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { defineConfig } from '@island.is/nest/config'
+
+export const FeatureFlagConfig = defineConfig({
+  name: 'FeatureFlag',
+  load: (env) => ({
+    sdkKey: env.required(
+      'CONFIGCAT_SDK_KEY',
+      'YcfYCOwBTUeI04mWOWpPdA/KgCHhUk0_k2BdiKMaNh3qA',
```
Comment (saevarma) [inline libs/nest/feature-flags/src/lib/feature-flag.config.ts:8]: Should the SDK key be in the source?

Comment (eirikurn) [reply libs/nest/feature-flags/src/lib/feature-flag.config.ts:8]: It's already in the codebase and included in all of our production browser bundles. So I'll leave this for now until we decide to remove it globally at some point.

**Review by pshomov — APPROVED** [review]
LGTM

**Review by saevarma — APPROVED** [review]
Great addition!

**Pull request merged**